content/200-concepts/100-components/03-prisma-migrate/010-mental-model.mdx

```diff
@@ -0,0 +1,178 @@
+---
+title: 'A mental model for Prisma Migrate'
+metaTitle: 'A mental model for Prisma Migrate'
+metaDescription: 'A mental model guide for working with Prisma Migrate in your project'
+tocDepth: 3
+---
+
+<TopBlock>
+
+This guide provides a conceptual overview of database migrations using Prisma Migrate when working with relational databases. It covers: what database migrations are, their value, and what Prisma Migrate is and how you can evolve your database schema with Prisma Migrate in different environments.
+
+**This guide does not apply to MongoDB** because Prisma Migrate only supports relational databases.
+
+</TopBlock>
+
+## What are database migrations?
+
+Database migrations are a controlled set of changes that modify and evolve the structure of your database schema. Migrations help you transition your database schema from one state to another. For example, within a migration you can create or remove tables and columns, split fields in a table, or add types and constraints to your database.
+
+### Patterns for evolving database schemas
+
+This section describes general schema migration patterns that exist for evolving database schemas.
+
+The two main schema migration patterns are:
+
+- **Model/Entity-first migration:** with this pattern, you define the structure of the database schema with code and then use a migration tool to generate the SQL, for example, for syncing your application and database schema.
+
+![](https://user-images.githubusercontent.com/33921841/194377029-305dc60b-f651-46e4-87f1-508884ebd972.png)
+
+- **Database-first migration:** with this pattern, you define the structure of your database and apply it to your database using SQL. You then _introspect_ the database to generate the code that describes the structure of your database to sync your application and database schema.
+
+![](https://user-images.githubusercontent.com/33921841/194378557-a18d5ad3-96a8-45da-8377-e078893ef2bb.png)
```

Comment:
One of the graphics has dots in the background, the other one does not.

Reply:
Ah, that was unintentional. I didn't notics that. I'll fix that

Reply:
I looked into this and turns out the dots are there.... it's just that the image is really big and the dots are invisible. I'll go ahead and resolve this conversation though because I'll create a design ticket for the images to be prettified